library/Director/Objects/IcingaServiceSet.php

```diff
@@ -327,4 +327,29 @@ protected function beforeStore()
             );
         }
     }
+
+    public function onSingleIcingaConfig(IcingaConfig $config)
+    {
+        parent::onSingleIcingaConfig($config);
+
+        if ($this->id !== null) {
```

Comment:
Please be careful, `$this->id` is a very special beast, it will probably end up calling `getId()`. It might work here, but I'd prefer to see the more explicit `get('id')` everywhere. This is part of some black magic in `DbObject` I absolutely do not like, I wrote id long long time ago - it scares me ;-)